api/app/views/spree/api/orders/_order.json.jbuilder

```diff
@@ -3,7 +3,7 @@
 json.cache! [I18n.locale, order] do
   json.(order, *order_attributes)
   json.display_item_total(order.display_item_total.to_s)
-  json.total_quantity(order.line_items.sum(:quantity))
+  json.total_quantity(order.line_items.to_a.sum(&:quantity))
```

Comment:
The order should already have preloaded the line items. If we do a `sum` in the database than we always load it from the database. And even if the line items have not been preloaded yet, they would be loaded by this call. It is always preferable to use array methods instead of `where` statements as the latter will always - no matter what - load from the database, while the former would only load if necessary.

Reply:
Great write up https://www.speedshop.co/2019/01/10/three-activerecord-mistakes.html

Reply:
Oh, I see:

```
order = create :order_with_line_items, line_items_count: 10

Benchmark.bm do |bm|
  bm.report { order.line_items.sum :quantity }
  bm.report { order.line_items.to_a.sum &:quantity }
  bm.report { order.line_items.reload.to_a.sum &:quantity }
end

   user     system      total        real
0.000908   0.000217   0.001125 (  0.001124)
0.000036   0.000006   0.000042 (  0.000041)
0.000956   0.000162   0.001118 (  0.001122)
```
so, as long as the association has already been loaded from the DB, using ruby is so much faster... and when not, the numbers seem to be in the same ballpark. Then this makes definitely sense 👍